packages/core/tests/scalars.test.ts

```diff
@@ -0,0 +1,312 @@
+  name: 'PositiveInt',
+  serialize: (n) => n as number,
+  parseValue: (n) => {
+    if (typeof n !== 'number') {
+      throw new TypeError('Value must be a number');
+    }
+
+    if (n >= 0) {
+      return n;
+    }
+
+    throw new TypeError('Value must be positive');
+  },
+});
+
+enum Diet {
+  HERBIVOROUS,
+  CARNIVOROUS,
+  OMNIVORIOUS,
+}
+
+class Animal {
+  diet: Diet;
+
+  constructor(diet: Diet) {
+    this.diet = diet;
+  }
+}
+
+class Kiwi extends Animal {
+  name: string;
+  birthday: Date;
+  heightInMeters: number;
+
+  constructor(name: string, birthday: Date, heightInMeters: number) {
+    super(Diet.HERBIVOROUS);
+
+    this.name = name;
+    this.birthday = birthday;
+    this.heightInMeters = heightInMeters;
+  }
+}
+
+describe('scalars', () => {
+  it('when a scalar is added withScalars, the scalartype is added', () => {
+    const builder = new SchemaBuilder({}).withScalars({ PositiveInt: PositiveIntResolver });
+    builder.queryType();
+    builder.queryFields((t) => ({
+      positiveInt: t.field({
+        type: 'PositiveInt',
+        args: { v: t.arg.int({ required: true }) },
+        resolve: (_root, args) => args.v,
+      }),
+    }));
+
+    const schema = builder.toSchema();
+    expect(() =>
+      execute({
+        schema,
+        document: gql`query { positiveInt("hello") }`,
+      }),
+    ).toThrow('Expected Name, found String "hello"');
+  });
+
+  it('when scalars are added using withScalars, the Objects from the user schema are kept', async () => {
+    const builder = new SchemaBuilder<{
+      Objects: {
+        Example: { n: number };
+      };
+    }>({}).withScalars({
+      PositiveInt: PositiveIntResolver,
+    });
+
+    const Example = builder.objectType('Example', {
+      fields: (t) => ({
+        n: t.expose('n', {
+          type: 'PositiveInt',
+        }),
+      }),
+    });
+
+    builder.queryType({
+      fields: (t) => ({
+        example: t.field({
+          type: Example,
+          resolve: () => ({ n: 1 }),
+        }),
+      }),
+    });
+    const schema = builder.toSchema();
+
+    const result = await execute({
+      schema,
+      document: gql`
+        query {
+          example {
+            n
+          }
+        }
+      `,
+    });
+    expect(result.data).toEqual({ example: { n: 1 } });
+  });
+
+  it('when scalars are added using withScalars, the Interfaces from the user schema are kept', async () => {
+    const builder = new SchemaBuilder<{
+      Interfaces: {
+        Animal: Animal;
+      };
+    }>({}).withScalars({
+      NonNegativeInt: NonNegativeIntResolver,
+    });
+
+    builder.enumType(Diet, { name: 'Diet' });
+
+    builder.interfaceType('Animal', {
+      fields: (t) => ({
+        diet: t.expose('diet', {
+          exampleRequiredOptionFromPlugin: true,
+          type: Diet,
+        }),
+      }),
+    });
+
+    builder.objectType(Kiwi, {
+      name: 'Kiwi',
+      interfaces: ['Animal'],
```

Comment:
Just noticed you are based in New Zealand, which is where I'm currently on vacation 🥝

Reply:
New Zealand is an wonderful country to be in. I hope you're having a fantastic time exploring and experiencing everything it has to offer. Enjoy your stay! 🥝